friends/src/components/CreateFriendForm.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+
+class FriendForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      age: "",
+      email: ""
+    };
+  }
+
+  inputChange(event) {
+    const loc = event.target.name;
+    const val = event.target.value;
+    const newState = this.state;
+    newState[loc] = val;
+    this.setState(newState);
```

Comment:
Interesting approach! More commonly you might see something like this:
```
this.setState({
[event.target.name]: event.target.value
})
```
A bit less code, but does the same thing. Both work!